src/Certificate/AzureKeyVaultCertificateLocator.php

```diff
@@ -0,0 +1,134 @@
+        $certificate = $this->extractCertificateFromStoreData($certificateStoreData);
+
+        return $this->getAbsoluteTmpPathByContent($certificate);
+    }
+
+    /**
+     * Returns certificate store data from a base64 encoded azure key vault secret value.
+     *
+     * @param string $secretValue base64 encoded secret value.
+     *
+     * @return array the certificate store data.
+     *
+     * @throws AzureKeyVaultCertificateLocatorException
+     */
+    private function getCertificateStoreDataFromSecret(string $secretValue): array
+    {
+        $decodedSecretValue = base64_decode($secretValue);
+        if (!$decodedSecretValue) {
+            throw new AzureKeyVaultCertificateLocatorException('Could not decode secret from key vault.');
+        }
+
+        $certificateStoreData = [];
+        $passphrase = $this->hasPassphrase()
+            ? $this->getPassphrase()
+            : null
+        ;
+        if (!openssl_pkcs12_read($decodedSecretValue, $certificateStoreData, $passphrase)) {
+            throw new AzureKeyVaultCertificateLocatorException('Could not read certificate.');
+        }
+
+        return $certificateStoreData;
+    }
+
+    /**
+     * Extracts certificate data from certificate store data generated by openssl_pkcs12_read().
+     *
+     * @param array $certificateStoreData certificate store data generated by openssl_pkcs12_read().
+     *
+     * @return string certificate in pem format.
+     *
+     * @throws AzureKeyVaultCertificateLocatorException
+     */
+    private function extractCertificateFromStoreData(array $certificateStoreData): string
+    {
+        if (empty($certificateStoreData['pkey'])) {
+            throw new AzureKeyVaultCertificateLocatorException('Missing pkey in certificate store data.');
+        }
+
+        if (empty($certificateStoreData['cert'])) {
+            throw new AzureKeyVaultCertificateLocatorException('Missing cert in certificate store data.');
+        }
+
+        $combinedCertificate = $certificateStoreData['pkey']
+            . PHP_EOL
+            . $certificateStoreData['cert']
+        ;
+
+        if (array_key_exists('extracerts', $certificateStoreData)) {
+            $combinedCertificate .= is_array($certificateStoreData['extracerts'])
+                ? implode('', $certificateStoreData['extracerts'])
+                : $certificateStoreData['extracerts']
+            ;
+        }
+
+        return $combinedCertificate;
+    }
+
+    /**
+     * Creates a temporary file with the provided content and returns the absolute path to the temporary file.
+     * The file will be removed from the filesystem when no more references exists to the file.
+     *
+     * @param string $content the content of the temporary file.
+     *
+     * @return string the absolute path to the temporary file.
+     */
+    private function getAbsoluteTmpPathByContent(string $content): string
+    {
+        static $tmpFile = null;
```

Comment:
When static it will not be GC'ed between calls? But when always overriden in the next line... it should not be static.

Reply:
👍

Reply:
Fixed in 44facae